app/code/Magento/Catalog/Test/Unit/ViewModel/Product/Checker/AddToCompareAvailabilityTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Catalog\Test\Unit\ViewModel\Product\Checker;
+
+use Magento\Catalog\ViewModel\Product\Checker\AddToCompareAvailability;
+use Magento\CatalogInventory\Api\StockConfigurationInterface;
+use Magento\Catalog\Model\Product;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use PHPUnit\Framework\MockObject\MockObject;
+
+/**
+ * Unit test for Magento\Catalog\ViewModel\Product\Checker\AddToCompareAvailability.
+ */
+class AddToCompareAvailabilityTest extends \PHPUnit\Framework\TestCase
+{
+
+    /**
+     * @var AddToCompareAvailability
+     */
+    private $viewModel;
+
+    /**
+     * @var StockConfigurationInterface|MockObject 
+     */
+    protected $stockConfiguration;
+
+    /**
+     * @var ObjectManager
+     */
+    private $objectManager;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp(): void
+    {
+
+        $this->stockConfiguration =
+            $this->getMockBuilder(StockConfigurationInterface::class)
+            ->getMock();
+
+        $this->viewModel = $this->getObjectManager()->getObject(
+            AddToCompareAvailability::class,
+            [
+                'stockConfiguration' => $this->stockConfiguration
+            ]
+        );
+    }
+
+    /**
+     * Test IsAvailableForCompare() with data provider
+     *
+     * @param bool $status
+     * @param bool $isSalable
+     * @param array $isInStock
+     * @param bool $isShowOutOfStock
+     * @return boolean
+     * @dataProvider isAvailableForCompareDataProvider
+     */
+    public function testIsAvailableForCompare($status, $isSalable, $isInStock, $isShowOutOfStock): bool
+    {
+        $productMock = $this->getMockBuilder(Product::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $productMock->expects($this->once())
+            ->method('getStatus')
+            ->willReturn($status);
+
+        $productMock->expects($this->any())
+            ->method('isSalable')
+            ->willReturn($isSalable);
+
+        $productMock->expects($this->any())
+            ->method('getQuantityAndStockStatus')
+            ->willReturn($isInStock);
+
+        $this->stockConfiguration->expects($this->any())
+            ->method('isShowOutOfStock')
+            ->willReturn($isShowOutOfStock);
+
+        return $this->viewModel->isAvailableForCompare($productMock);
```

Comment:
Why are you returning anything from the test?

Reply:
@lbajsarowicz Thanks for feedback. Actually `isShowOutOfStock` returns bool, So I passed data from data provider with `true` & `false`. Also `ScopeConfigInterface` not injected in original file that's why I am not used `isSetFlag`. correct me if I am wrong.

Reply:
There is no need to return anything from Test Method. Unit Tests are void.

In this case, what you are doing is state-based testing (based on the actions that are performed, not the result that is asserted).

Reply:
@lbajsarowicz Thanks for feedback, I have update with assert. can you please review.

Reply:
Now is way better!

Reply:
@lbajsarowicz Thanks for valuable feedback.